### PR TITLE
let sbatch #PBS parser pick up on '#PBS -d' in job script

### PR DESCRIPTION
**Pull request by boegel**
@itkovian @stdweird I'm not sure this is complete, but it looks like it could be (at least it compiles, according to Travis CI...). ;)

Any suggestions on how to test whether this is complete or not, or how to proceed from here?

### CODE REVIEW

File: src/sbatch/opt.c [hunk 1]
```diff
@@ -2271,13 +2271,14 @@ static void _set_pbs_options(int argc, char **argv)
...
 	char *sep = "";
-	char *pbs_opt_string = "+a:A:c:C:e:hIj:J:k:l:m:M:N:o:p:q:r:S:t:u:v:VW:z";
+	char *pbs_opt_string = "+a:A:c:C:d:e:hIj:J:k:l:m:M:N:o:p:q:r:S:t:u:v:VW:z";
 
 	struct option pbs_long_options[] = {
 		{"start_time", required_argument, 0, 'a'},
 		{"account", required_argument, 0, 'A'},
 		{"checkpoint", required_argument, 0, 'c'},
 		{"working_dir", required_argument, 0, 'C'},
+		{"working_dir", required_argument, 0, 'd'},
```
Comment (stdweird) [inline src/sbatch/opt.c]: why is the `working_dir` the same as previous option?

**Comment by stdweird** [comment]
compile it somewhere, and try the sbatch binary.
or loet @itkovian make an rpm (`clusterbuildrpm -Q` can probably do that for you)

File: src/sbatch/opt.c (hunk 1 above)
Comment (boegel) [reply src/sbatch/opt.c]: Well, for one, it's not used at all, as far as I can tell (see the `case 'C': break;` below).

Also, `qsub -C` is something else, as far I can tell. From the man page:

```
       -C directive_prefix
               Defines the prefix that declares a directive to the qsub command within the script file.
               See the paragraph on script directives in the Extended Description section.

               If the -C option is presented with a directive_prefix argument that is the null string,
               qsub will not scan the script file for directives.
```

I think that's for changing the `#PBS` that's picked up in job scripts with something else, not related to "working directory" at all.

**Comment by itkovian** [comment]
@hajgato suggested to drop the PBS plugin at some point, so not dure if this is still applicable.

**Comment by boegel** [comment]
This can be closed, since:

* this can be handled better in the `qsub` wrapper or submitfilter
* there's an easy workaround for this, users can use `cd <dir>` or `cd $PBS_O_WORKDIR` in their job script instead

**Pull request closed without merging**